planner/core/exhaust_physical_plans.go

```diff
@@ -1264,9 +1300,38 @@ func (la *LogicalAggregation) getHashAggs(prop *property.PhysicalProperty) []Phy
 }
 
 func (la *LogicalAggregation) exhaustPhysicalPlans(prop *property.PhysicalProperty) []PhysicalPlan {
-	aggs := make([]PhysicalPlan, 0, len(la.possibleProperties)+1)
-	aggs = append(aggs, la.getHashAggs(prop)...)
-	aggs = append(aggs, la.getStreamAggs(prop)...)
+	preferHash := (la.preferAggType & preferHashAgg) > 0
+	preferStream := (la.preferAggType & preferStreamAgg) > 0
+	if preferHash && preferStream {
+		errMsg := "Optimizer aggregation hints are conflicted"
+		warning := ErrInternal.GenWithStack(errMsg)
+		la.ctx.GetSessionVars().StmtCtx.AppendWarning(warning)
+	}
+
+	hashAggs := la.getHashAggs(prop)
+	if hashAggs != nil && preferHash && !preferStream {
+		return hashAggs
+	}
+
+	streamAggs := la.getStreamAggs(prop)
+	if streamAggs != nil && preferStream && !preferHash {
+		// The following three lines are used to cover a bug related to possibleProperties.
+		for i := range streamAggs {
+			for j := range streamAggs[i].(*PhysicalStreamAgg).childrenReqProps {
+				streamAggs[i].(*PhysicalStreamAgg).childrenReqProps[j].Enforced = true
```

Comment:
I think we can remove this line since we have set `Enforced` to `true` in `getEnforcedStreamAggs`.